sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java

```diff
@@ -343,9 +355,48 @@ public void channelDisconnected(ChannelHandlerContext ctx, ChannelStateEvent e)
     }
 
     private void handleStartupBody(ChannelBuffer buffer, Channel channel) {
-        session = readStartupMessage(buffer);
-        Messages.sendAuthenticationOK(channel);
+        sessionContext = readStartupMessage(buffer);
+        session = sqlOperations.createSession(sessionContext);
+        if (authService.enabled()) {
+            authenticate(channel);
+        } else {
+            Messages.sendAuthenticationOK(channel);
+            sendReadyForQuery(channel);
+        }
+    }
+
+    private void authenticate(Channel channel) {
+        InetAddress address = ((InetSocketAddress) channel.getRemoteAddress()).getAddress();
+        AuthenticationMethod authMethod = authService.resolveAuthenticationType(sessionContext.userName(), address);
+        if (authMethod == null) {
```

Comment:
Did you consider having a default implementation of AuthenticationMethod for this case as well?